src/main/scala/com/kodekutters/stix/Extensions.scala

```diff
@@ -18,84 +16,75 @@ import Util._
   */
-case class ArchiveFileExt(contains_refs: Option[List[String]] = None,
+@JsonCodec case class ArchiveFileExt(contains_refs: Option[List[String]] = None,
                           version: Option[String] = None,
                           comment: Option[String] = None) extends Extension
 
 object ArchiveFileExt {
   val `type` = "archive-ext"
-  implicit val fmt = Json.format[ArchiveFileExt]
 }
 
 /**
   * The Alternate Data Stream type represents an NTFS alternate data stream.
   */
-case class AlternateDataStream(name: String,
+@JsonCodec case class AlternateDataStream(name: String,
                                hashes: Option[Map[String, String]] = None,
                                size: Option[Long] = None)
 
-object AlternateDataStream {
-  implicit val fmt = Json.format[AlternateDataStream]
-}
-
 /**
   * The NTFS file extension specifies a default extension for capturing properties specific to the storage of the file on the NTFS file system.
   */
-case class NTFSFileExt(sid: Option[String] = None,
+@JsonCodec case class NTFSFileExt(sid: Option[String] = None,
                        alternate_data_streams: Option[List[AlternateDataStream]] = None) extends Extension
 
 object NTFSFileExt {
   val `type` = "ntfs-ext"
-  implicit val fmt = Json.format[NTFSFileExt]
 }
 
 /**
   * The PDF file extension specifies a default extension for capturing properties specific to PDF files.
   */
-case class PdfFileExt(version: Option[String] = None,
+@JsonCodec case class PdfFileExt(version: Option[String] = None,
                       is_optimized: Option[Boolean] = None,
                       document_info_dict: Option[Map[String, String]] = None,
                       pdfid0: Option[String] = None,
                       pdfid1: Option[String] = None) extends Extension
 
 object PdfFileExt {
   val `type` = "pdf-ext"
-  implicit val fmt = Json.format[PdfFileExt]
 }
 
 /**
   * The Raster Image file extension specifies a default extension for capturing properties specific to raster image files.
   */
-case class RasterImgExt(image_height: Option[Long] = None,
-                        image_width: Option[Long] = None,
-                        bits_per_pixel: Option[Long] = None,
-                        image_compression_algorithm: Option[String] = None,
-                        exif_tags: Option[Map[String, Either[Long, String]]] = None) extends Extension
-
-object RasterImgExt {
-  val `type` = "raster-image-ext"
-  implicit val fmt = Json.format[RasterImgExt]
-}
+//@JsonCodec case class RasterImgExt(image_height: Option[Long] = None,
+//                        image_width: Option[Long] = None,
+//                        bits_per_pixel: Option[Long] = None,
+//                        image_compression_algorithm: Option[String] = None,
+//                        exif_tags: Option[Map[String, Either[Long, String]]] = None) extends Extension
```

Comment:
As far as I remember, this was mainly prior to have a decoder/encoder for `Either[Long, String]`.